client/blocks/edit-gravatar/index.jsx

```diff
@@ -158,49 +155,41 @@ export class EditGravatar extends Component {
 		}
 	}
 
+	renderInstructionText() {
+		return (
+			<div className="edit-gravatar__label-container">
+				<Gridicon icon="cloud-upload" size={ 36 } />
+				<span className="edit-gravatar__label">
+					{ this.props.translate( 'Click to change photo' ) }
+				</span>
+			</div>
+		);
+	}
+
 	render() {
 		const {
-			isOffline: userIsOffline,
 			isUploading,
-			translate,
 			user
 		} = this.props;
 		return (
-			<div>
+			<div className="edit-gravatar">
 				{ this.renderImageEditor() }
-				<FormLabel>
-					{
-						translate( 'Avatar', {
-							comment: 'A section heading on the profile page.'
-						} )
-					}
-				</FormLabel>
-				<div
-					className={
-						classnames( 'edit-gravatar__image-container',
-							{ 'is-uploading': isUploading }
-						)
-					}
-				>
-					<Gravatar
-						imgSize={ 270 }
-						size={ 100 }
-						user={ user }
-					/>
-					{ isUploading && <Spinner className="edit-gravatar__spinner" /> }
-				</div>
-				<p>
-					{
-						translate( 'To change, select an image or ' +
-							'drag and drop a picture from your computer.' )
-					}
-				</p>
 				<FilePicker accept="image/*" onPick={ this.onReceiveFile }>
-					<Button
-						disabled={ userIsOffline || isUploading || ! user.email_verified }
+					<div
+						className={
+							classnames( 'edit-gravatar__image-container',
+								{ 'is-uploading': isUploading }
+							)
+						}
 					>
-						{ translate( 'Select Image' ) }
-					</Button>
+						<Gravatar
+							imgSize={ 400 }
+							size={ 150 }
+							user={ user }
+						/>
+						{ ! isUploading && this.renderInstructionText() }
```

Comment:
The modularization is nice, but we usually prefer to keep the JSX explicit inside of `render()` for the sake of readability: "what actually is being rendered?"

If we get to the point where the modularization is really important we should probably consider creating an altogether new React component out of it.